### PR TITLE
heap: add heap_reset function

### PR DESCRIPTION
**Pull request by hstern**

### CODE REVIEW

File: heap.c [hunk 1]
```diff
@@ -100,6 +110,20 @@ siftup(struct heap *h, size_t pos)
 }
 
 void
+heap_heapify(struct heap *h)
+{
+	for (ssize_t i=ptrvec_size(h->vec)/2-1; i>=0; i--) {
```
Comment (edmonds) [inline heap.c]: `ptrvec_size()` returns a `size_t`, not a `ssize_t`, so use `size_t` here. Also please add some spaces. So:

``` c
for (size_t i = ptrvec_size(h->vec) / 2 - 1; i >= 0; i--) {
```

File: heap.c [hunk 2]
```diff
@@ -100,6 +110,20 @@ siftup(struct heap *h, size_t pos)
...
+		siftup(h, i);
```
Comment (edmonds) [inline heap.c:116]: Does the return value of `siftup()` need to be checked?

File: heap.c (hunk 1 above)
Comment (hstern) [inline heap.c]: however, a size_t/2 can never overflow a ssize_t.  I can rewrite as (size_t i=size/2; i-- > 0; ) if you really want.

File: heap.c (hunk 2 above)
Comment (hstern) [inline heap.c:116]: we don't check it anywhere else in heap.c and we would have to change all of the function signatures to return the error code (they are void).  do you want to do this?

Comment (edmonds) [inline heap.c:116]: I was mainly wondering if the algorithm can fail, are we ignoring the error return? If it's just a matter of propagating the return value or not, and the algorithm always succeeds, then it's OK.

**Pull request merged**